Slim/Control/Queries.pm

```diff
@@ -792,9 +794,31 @@ sub albumsQuery {
 			$request->addResultLoopIfValueDefined($loopname, $chunkCount, 'work_id', $c->{'tracks.work'});
 			$request->addResultLoopIfValueDefined($loopname, $chunkCount, 'work_name', $c->{'works.title'});
 			$request->addResultLoopIfValueDefined($loopname, $chunkCount, 'composer', $c->{'composer.name'});
-			$request->addResultLoopIfValueDefined($loopname, $chunkCount, 'grouping', $c->{'tracks.grouping'});
+			$request->addResultLoop($loopname, $chunkCount, 'grouping', $c->{'tracks.grouping'}||"");
+
+			# If we're dealing here with a Work rather than the full album, add relevant information into the album name and set favorites URL for the work. Moved to here
+			# from BrowseLibrary.pm so that it applies to all UIs.
+			my $albumTitle = $c->{'composer.name'} ? $c->{'composer.name'} . cstring($client, 'COLON') . ' ' : '';
+			if ( $c->{'tracks.work'} ) {
+				$albumTitle .= $c->{'works.title'} . ' (';
+				$albumTitle .= "$c->{'tracks.grouping'} " if $c->{'tracks.grouping'};
+				$albumTitle .= cstring($client,'FROM') . ' ';
+			}
```

Comment:
I'm sorry, that's UI code. It doesn't belong here, IMHO. If you check the codebase in this file you'll hardly find any UI elements or even strings. It's really something that different UIs probably want to handle differently. 

The queries should be about the data, not their presentation.

Reply:
It isn't UI code: it's a consequence of reusing albumsQuery for Works - at this level, the work is a subset of the album, so the title of the "album" (album-work?) needs to reflect that. 

An "album-work" is a virtual entity, which shares almost everything with the Album entity, except the title. So what this is doing is overriding the album title when we're reusing the albumsQuery to provide a list of album-works.

This is definitely not presentation - it is a string describing an instance of a work, in exactly the same way as albums.title does for an instance of an album or tracks.title does for an instance of a track.

I could move this code into a separate function, in Schema or somewhere, but it would still need to be called from here.

Looking back, maybe I could have created an album_work table to reflect properly the many-to-many relationship between album and work, and a new query albumWorkQuery, which would be called instead of albumsQuery whenever work_id was passed. But that would lead to a lot of duplication of logic, and would still need to use string processing unless we added language-specific strings to the database ("COLON") which would be a very bad idea.

Reply:
But this is creating "Composer: Work from Album", isn't it? What if the UI wanted to have those pieces of information on separate lines? Or "Work (Composer - Album)" etc.? Keep in mind that the "old" player UI gives you a ton of flexibility formatting the information in the way you (the user) like. But here you are taking away this UI decision in the database query code. 

We should return all information as their own tags an leave it to the UI code to format it the way they want it to be. The composer and work names are/could be part of the result set, right?

Reply:
I am still considering the idea of Work Title formatting options, like the existing Title Format in Settings -- Interface. But as you know, Material doesn't use that. So if we do this, it would still, in my view, be necessary to use this in albumsQuery, to force consistency of whatever the user chooses across all UIs. And I don't think this user config option is necessary for this PR to be useful.

And in the end, if you think of a work as a sub-album, it needs a meaningful title which, however formatted, must include the Composer and Work (Work names on their own are rarely meaningful - think how many "Symphony No. 1"s are out there). And the album name (and grouping text, if present) are absolutely necessary to disambiguate in the case of multiple versions of the same work in the user's library. It's not optional.